examples/p4est_2d_dgsem/elixir_advection_restart_amr.jl

```diff
@@ -0,0 +1,54 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# create a restart file
+
+elixir_file = "elixir_advection_extended.jl"
+restart_file = "restart_000021.h5"
+
+trixi_include(@__MODULE__, joinpath(@__DIR__, elixir_file))
+
+###############################################################################
+# adapt the parameters that have changed compared to "elixir_advection_extended.jl"
+
+# Note: If you get a restart file from somewhere else, you need to provide
+# appropriate setups in the elixir loading a restart file
+
+restart_filename = joinpath("out", restart_file)
+mesh = load_mesh(restart_filename)
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
+                                    boundary_conditions = boundary_conditions)
+
+tspan = (load_time(restart_filename), 2.0)
+dt = load_dt(restart_filename)
+ode = semidiscretize(semi, tspan, restart_filename);
+
+# Do not overwrite the initial snapshot written by elixir_advection_extended.jl.
+save_solution.condition.save_initial_solution = false
+
+# Add AMR callback
+amr_controller = ControllerThreeLevel(semi, IndicatorMax(semi, variable = first),
+                                      base_level = 0,
+                                      med_level = 0, med_threshold = 0.8,
+                                      max_level = 1, max_threshold = 1.2)
+amr_callback = AMRCallback(semi, amr_controller,
+                           interval = 5,
+                           adapt_initial_condition = false,
+                           adapt_initial_condition_only_refine = true)
+callbacks_ext = CallbackSet(amr_callback, callbacks.discrete_callbacks...)
+
+integrator = init(ode, CarpenterKennedy2N54(williamson_condition = false),
+                  dt = dt, # solve needs some value here but it will be overwritten by the stepsize_callback
+                  save_everystep = false, callback = callbacks_ext, maxiters = 100_000);
```

Comment:
Why don't we use the `ode_default_options()` here as we do elsewhere? And do we need the `maxiters = 100_000`? I see that it is also in
https://github.com/trixi-framework/Trixi.jl/blob/76719a85e0226ba9883dcf3817c6afe8daffb3f6/examples/tree_2d_dgsem/elixir_advection_restart.jl#L32
but I am also wondering if we really need that there.

Reply:
Thanks for the hints!

I just copied `elixir_advection_restart.jl` and then forgot to check this `maxiters` parameter. This example is not run with coverage, so I think you are right and it is unneeded.

Concerning `ode_default_options()`: this would replace `save_everystep = false`, right? This I could just change it.

Reply:
> I just copied elixir_advection_restart.jl and then forgot to check this maxiters parameter. This example is not run with coverage, so I think you are right and it is unneeded.

Does it matter whether it is run with coverage or not? Isn't `maxiters` always overwritten by 1 in the coverage tests except you "reoverwrite" it by explicitly passing `coverage_override = (maxiter = ...,)` as you also do here in the test? But I think it still makes sense to keep it, see [this discussion I just found](https://github.com/trixi-framework/Trixi.jl/pull/1737/files#r1394256240)

> Concerning ode_default_options(): this would replace save_everystep = false, right? Then I could just change it.

Yes, it would. On the other hand, we usually do not use it in elixirs where we used CFL-based timestepping rather than error-based timestepping, see https://trixi-framework.github.io/Trixi.jl/stable/tutorials/time_stepping/#adaptive_step_sizes. I actually think, we should remove the `ode_default_options()` from `elixir_advection_restart.jl`. This was introduced later in https://github.com/trixi-framework/Trixi.jl/pull/1565 rather than in https://github.com/trixi-framework/Trixi.jl/pull/1384 and looks spurious to me. Then we should also remove it in https://github.com/trixi-framework/Trixi.jl/blob/3b52a3013fb09564240a3bc876c5e91f5f6c8642/examples/tree_2d_dgsem/elixir_advection_extended.jl#L81 (especially since there _additionally_ is `save_everystep = false`, which is already included in `ode_default_options()...`).